zio-query/shared/src/main/scala/zio/query/ZQuery.scala

```diff
@@ -1345,26 +1336,28 @@ object ZQuery {
                 ZIO.succeed(
                   Result.blocked(
                     BlockedRequests.single(dataSource, BlockedRequest(request, promise)),
-                    Continue[R, E, A, B](promise)
+                    Continue(promise)
                   )
                 )
               case Right(promise) =>
                 promise.poll.flatMap {
-                  case None     => ZIO.succeed(Result.blocked(BlockedRequests.empty, Continue[R, E, A, B](promise)))
+                  case None     => ZIO.succeed(Result.blocked(BlockedRequests.empty, Continue(promise)))
                   case Some(io) => io.exit.map(Result.fromExit)
                 }
             }
             ZQuery.currentCache.getWith {
-              case cache: Cache.Default => foldPromise(cache.lookupUnsafe(request, fiberId)(Unsafe.unsafe, implicitly))
+              case cache: Cache.Default => foldPromise(cache.lookupUnsafe(request, fiberId)(Unsafe.unsafe))
               case cache                => cache.lookup(request).flatMap(foldPromise)
             }
-          } else
-            Promise.makeAs[E, B](fiberId).map { promise =>
+          } else {
+            ZIO.succeed {
```

Comment:
Out of curiosity, why this change? Isn't `Promise.makeAs` doing the same thing?

Reply:
Because when using `Promise.makeAs` we have to use `.map` afterwards (which is an extra flatMap). Since this is run for every `ZQuery.fromRequest`, I think it's worth to use the `unsafe` method in order to reduce the overall number of flatMaps in the query

Reply:
Ah right, I missed that